File: python/fizzbuzz/fizzbuzz.py
"""A FizzBuzz program"""

# import necessary supporting libraries
from numbers import Number

def fizz(x):
  """
    Checks to see if the input `x` is numeric and a multiple of 3.
    If it is, it outputs 'Fizz'.
    Otherwise, it outputs the input.
  """
  return 'Fizz' if isinstance(x, Number) and x % 3 == 0 else x

def buzz(x):
  """
    Checks to see if the input `x` is numeric and a multiple of 5.
    If it is, it outputs 'Buzz'.
    Otherwise, it outputs the input.
  """
  return 'Buzz' if isinstance(x, Number) and x % 5 == 0 else x

def fibu(x):
  """
    Checks to see if the input `x` is numeric and a multiple of 15.
    If it is, it outputs 'FizzBuzz'.
    Otherwise, it outputs the input.
  """
  return 'FizzBuzz' if isinstance(x, Number) and x % 15 == 0 else x

def play(start, end):
  """
    Given a start number and an end number, play FizzBuzz
    starting at `start` and ending at `end` and output
    the results as a list (array).
  """
  # initialize an empty collection to hold our output
  output = []
  # loop through the numbers from start to end
  for x in range(start, end + 1):
    output.append(buzz(fizz(fibu(x))))
  # return the output
  return output

File: js/fizzbuzz/fizzbuzz.js

const fizz = function (x) {
  // check to see if x exists
  if (typeof x !== 'undefined') {
    // yes, x exists, carry on
    return typeof x === 'number' && x % 3 === 0 ? 'Fizz' : x
  } else {
    // no, x does not exist, throw an error
    throw new Error('there is no input')
  }
}

const buzz = function (x) {
  // check to see if x exists
  if (typeof x !== 'undefined') {
    // yes, x exists, carry on
    return typeof x === 'number' && x % 5 === 0 ? 'Buzz' : x
  } else {
    // no, x does not exist, throw an error
    throw new Error('there is no input')
  }
}

const fibu = function (x) {
  // check to see if x exists
  if (typeof x !== 'undefined') {
    // yes, x exists, carry on
    return typeof x === 'number' && x % 15 === 0 ? 'FizzBuzz' : x
  } else {
    // no, x does not exist, throw an error
    throw new Error('there is no input')
  }
}

const play = function (start, end) {
  // declare an empty array variable to hold our output
  const output = []
  // loop through the numbers from start to end
  for (let x = start; x <= end; x++) {
    output.push(buzz(fizz(fibu(x))))
  }
  return output
}

module.exports = {
  fizz,
  buzz,
  fibu,
  play
}
